client/utils.c

```diff
@@ -219,6 +219,51 @@ TDNFUtilsMakeDirs(
     goto cleanup;
 }
 
+uint32_t
+TDNFIsFileOrSymlink(
```

Comment:
Is this function (and it's check) really needed? You can always use `realpath()` to convert symlinks to absolute paths very early and then work only with absolute paths and URLs.

Reply:
Point of this is to find out if the cmd line argument is a file (or a symlink), in contrast to a simple package or a URL. The real path is not interesting at this moment. Also, if the symlink doesn't resolve we shouldn't fail at this point, because if the link exists the user's intention is most likely to use this as the package path to be installed. They will get the error later, but if we interpret this as a plain package from a repo this will be more confusing.